### DISCUSSION TITLE
Having trouble loading raster data into flopy

### DISCUSSION THREAD

**Discussion by AH-CSM** [modflowpy/flopy#1501] [Q&A] [answered]
Has anyone been using spyder or Anaconda to run flopy and run into the issue of not being able to load raster data into the model for a unique grid? My personal struggle is that I am using the sample code suggested by @jlarsen-usgs found at https://github.com/modflowpy/flopy/blob/develop/examples/Notebooks/flopy3_raster_intersection.ipynb

However, when I attempt to run my code I am receiving the output:

"InuputError: Missing optional dependency 'rasterio'. Use pip or conda to install rasterio."

I have followed these suggestions, as well as downloaded it from the Anaconda.org IOOS page directly, and many other suggestions floating around the internets and I am still receiving this error.
Any thoughts or suggestions?
Thank you in advance!

**Comment by vcantarella** [comment]
rasterio is the library used inside flopy to handle raster data. If you use anaconda, I suggest you install it inside your flopy environment following the recommendations from their website:

[https://github.com/conda-forge/rasterio-feedstock](https://github.com/conda-forge/rasterio-feedstock)

**Reply by AH-CSM** [reply]
Yes, I tried that and I have checked the versions available and they populate up in the command window as being in the library but I still get this error.

**Reply by jlarsen-usgs** [reply]
@AH-CSM 
If you're having trouble installing rasterio and working on a windows machine, I recommend downloading pre-compiled binaries for GDAL, FIONA, and the rasterio libraries. These binary packages can be found here (https://www.lfd.uci.edu/~gohlke/pythonlibs/). Make sure that the number after "cp" in the filename matches the version of python you are using. Example python 3.9, "cp39". Also pay attention to 64 vs 32 bit binaries. My guess is that you'll be working with 64 bit. 

Once these wheels are downloaded, they can be installed by using:
```
pip install <wheel name>
```

**Reply by mwtoews** [reply]
@jlarsen-usgs rasterio has binary wheels for Windows for Python versions 3.8+ on [PyPI](https://pypi.org/project/rasterio/#files), so a `pip install rasterio` command on Windows should now work. But more importantly, mixing binary wheels from pip in a conda environment might encounter unexpected behaviors. It's generally advised that conda users should install conda packages when possible.

**Comment by mwtoews** [comment, marked as answer]
Make sure you install `rasterio` to the correct conda [environment](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#activating-an-environment). You can check the environment from your Python console with:
```python
import sys
print(sys.exec_prefix)
```
and check to see if it's the same as the console where you typed (e.g.) `conda install rasterio` with:
```bash
python -c "import sys; print(sys.exec_prefix)"
```
if it's different, then you need to activate the correct environment to install it for use.
